### PR TITLE
feat(wasm): Import/export account

### LINKED ISSUE #816: feat(import): wire in Import account in UplinkWeb
As of now it is not possible to import account in the app. Since this is an existing feature in warp we can wire this feature in to make sure users can import old accounts

### PR DESCRIPTION
**Pull request by Flemmli97**
### What this PR does 📖

- Adds ability to import/export accounts
- Importing:
  - At login screen: Clicking on import brings up pin page to type in a new pin.
  - Continuing brings up a passphrase page where the user can either type the passphrase in or upload from a file. Words in the file can be separated by either a newline or whitespace
  - Importing can be done either via remote (dont think its enabled yet) or from a file
- Exporting:
  - Adds ability to export in the profile settings page. Either to remote or to a file. If remote is enabled at some point that should also be done automatically. The button is there if someone wants to manually do it too

### Which issue(s) this PR fixes 🔨

-   Resolve #816

### Special notes for reviewers 🗒️

A lot of errors are due to wasm update which is handled in #868. I just didnt include the stuff in that pr here too. So ideally merge #868 first

### CODE REVIEW

**Review by stavares843 — CHANGES_REQUESTED** [review]
go to import > dont import > back > sign up > add name > error

https://github.com/user-attachments/assets/68dd199e-38e4-441e-bad6-d30a56a49890



chrome, macOS

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2024-11-27, às 18 27 12


i made a new account
settings
export to both file and remote
deleted account
import 
add seed
use file

says identity does not exist - does that mean it was really removed?

**Review by stavares843 — CHANGES_REQUESTED** [review]
friend requests are not working on this branch


flow
- create new account
- add friend
- request does not appear

@phillsatellite and i tried it and we dont get the requests, works on dev

**Comment by Flemmli97** [comment]
> Captura de ecrã 2024-11-27, às 18 27 12
> i made a new account settings export to both file and remote deleted account import add seed use file
> 
> says identity does not exist - does that mean it was really removed?

does import still work? it might be because warp instances are still dangling around with some polling stuff. the way i tested it just open a new incognito tab and import there. that way all dangling data is gone

**Comment by stavares843** [comment]
>does import still work? 



when i delete account no

**Comment by phillsatellite** [comment]
Looks like Im getting an error similar to Sara's except only when importing account remotely 

When I use Import from File seems to be working well 

Remote:

https://github.com/user-attachments/assets/bebc585c-dd86-474c-927c-052bfcfe082d

Screenshot 2024-12-02 at 11 26 36 AM

Import:

https://github.com/user-attachments/assets/34df83a3-5465-4b8d-8762-1989f49a8c8b

**Comment by Flemmli97** [comment]
> Looks like Im getting an error similar to Sara's except only when importing account remotely
> 
> When I use Import from File seems to be working well
> 
> Remote:
> 
>  remote.mov 
> Screenshot 2024-12-02 at 11 26 36 AM
> Import:
> 
>  import.mov

remote atm i dont think is enabled

**Comment by Flemmli97** [comment]
remote should now work if your account was created with shuttle enabled (so after the recent commit).
shuttle address is temporary for testing

File: src/lib/wasm/WarpStore.ts
Comment (dariusc93) [inline src/lib/wasm/WarpStore.ts]: after testing is done i would advise putting this behind some type of dev flag or commenting it out prior to merging

**Comment by stavares843** [comment]
thank you, we will check

**Comment by phillsatellite** [comment]
@Flemmli97 Im getting some weird behavior when testing 

When I go to create a new account I cant get passed the pin page. It will make me type in the pin twice and on the second time it will just endlessly load. I will throw the errors it gives in console in this comment below.

**Steps to reproduce:**
- Click create account 
- Create Username 
- Type in pin 
- Forced to type in pin again 
- Endless loading 

**Screen recording & errors:**

Flow:

https://github.com/user-attachments/assets/ec92961f-b7ee-474a-b3cd-ab25641de8bc

Errors:

Screenshot 2024-12-10 at 2 15 46 PM

**Comment by stavares843** [comment]
same here @Flemmli97 

image

**Comment by luisecm** [comment]
> same here @Flemmli97
> 
> image

Same on my side and this is causing all several automated tests to fail 💀

File: src/lib/wasm/MultipassStore.ts
Comment (dariusc93) [inline src/lib/wasm/MultipassStore.ts:624]: wouldnt it be best to use `get(this.multipassWritable)` altogether?

**Comment by stavares843** [comment]
@Flemmli97 when we copy paste the seed in iOS, the words are getting capitalized

also on web, you cant copy paste the whole seed at once

can you take a look please?

**Comment by stavares843** [comment]
iOS capitalization is fixed 

missing fixing this

> also on web, you cant copy paste the whole seed at once when you copy paste it pastes the whole seed on the first word instead of filling out the 12 word seed

File: src/lib/wasm/WarpStore.ts
Comment (dariusc93) [inline src/lib/wasm/WarpStore.ts]:
```suggestion
        // TODO: Uncomment later or add a flag to enable or disable
        // config.enable_shuttle_discovery(["/ip4/138.197.62.183/tcp/7008/ws/p2p/12D3KooW9tAgS6kZPmEtfRiBHsYC46dN2hv6hT6gQ8bqxzWw7y5X"])
```

**Review by dariusc93 — APPROVED** [review]
lgtm. left a small comment and a suggestion

Though testing has been done by QA to confirm that importing and exporting does work, it is still a WIP to finish up other aspects of it as well as clearing up some additional bugs, hence why shuttle will be disabled by default for now while changes are being made. We could block this PR until it is done or we could merge it in with it disabled until. a dev flag is in place to enable or disable it. 


Thoughts @phillsatellite @stavares843 @Flemmli97 ?

File: src/lib/wasm/MultipassStore.ts
Comment (Flemmli97) [reply src/lib/wasm/MultipassStore.ts:624]: this is because atm during import a temp warp instance is created
i guess we could just remove using the stored instance though since that atm is never used? thoughts?

**Comment by stavares843** [comment]
> shuttle will be disabled by default


But that means no messages are available on the import

**Comment by stavares843** [comment]
if you use any other file instead of the seed one, you will get these mambojambo on the screen

image

flow
- use backup file to import as a seed
- appears like that

**Comment by phillsatellite** [comment]
The bug @stavares843 commented is now fixed. I put the blocked label until we get the remote import account working

**Review by stavares843 — APPROVED** [review]

**Pull request merged**
